controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -382,7 +406,7 @@ func (w *Workload) updateClusterConfiguration(ctx context.Context, mutator func(
 			}
 			configMap.Data[clusterConfigurationKey] = updatedData
 			if err := w.Client.Update(ctx, configMap); err != nil {
-				return errors.Wrap(err, "failed to upgrade the kubeadmConfigMap")
+				return errors.Wrapf(err, "failed to upgrade the kubeadmConfigMap with the following properties %s", strings.Join(updatedKeys, ","))
```

Comment:
Question: what kind of error is returned from the APIServer? Maybe collecting and printing the `updatedKeys` here is not required because it is already in the error.

Reply:
Various kinds of errors could be returned with generic update failures, like config map not found, format validation error etc. I wanted to keep the previous behavior where individual keys are mentioned in updated error messages.

Reply:
I think the individual keys are only relevant for those validation errors, and there we should already get back the failing path, so I would propose to remove this additional info here.

Reply:
I am also fine with removing the keys 👍  however it will change the current behavior where **intended** keys to be updated will not be part of the error message.